samza-core/src/main/java/org/apache/samza/table/caching/guava/GuavaCacheTable.java

```diff
@@ -69,36 +87,48 @@ public void put(K key, V value) {
     }
   }
 
+  @Override
+  public CompletableFuture<Void> putAsync(K key, V value) {
+    put(key, value);
+    return CompletableFuture.completedFuture(null);
+  }
+
   @Override
   public void putAll(List<Entry<K, V>> entries) {
     entries.forEach(e -> put(e.getKey(), e.getValue()));
   }
 
+  @Override
+  public CompletableFuture<Void> putAllAsync(List<Entry<K, V>> entries) {
+    putAll(entries);
```

Comment:
This should happen within the future

Reply:
same comment as deleteAsync in LocalStoreBackedReadWriteTable.